src/streamLayer.js

```diff
@@ -22,7 +24,12 @@ var StreamFeatureLayer = L.esri.FeatureLayer.extend({
         if (response.timeInfo.trackIdField) {
           this.idField = response.timeInfo.trackIdField;
         }
-        this.streamUrl = response.streamUrls[0].urls[0];
+        this.streamUrl = response.streamUrls[0].urls
+          .filter(function filterUrl(url) {
+            // returns wss if in list, otherwise ws (if in list)
+            return (!!isWss ? url.match(/wss\:\/\//) : url.match(/ws\:\/\//)) !== null;
+          })
+          .filter(function(_, i) {return i === 0;}).shift(); // first available element only
```

Comment:
I think we can probably tidy this chunk up a bit. 

I would expect ArcGIS Server would use those indexes consistently so we could perhaps do something like... 
````
 this.streamUrl = this.options.wss ? response.streamUrls[0].urls[1] : response.streamUrls[0].urls[0];
````
Although the REST docs aren't particularly clear and I'm having trouble finding a new sample service to try...

Reply:
Hi again

Although I can see it brakes with style of the rest of the file, I choose the filter-array-implementation as it is a bit safer since it's indifferent to the order of the url-streams. I don't  know whether ArcGIS (or any other provider) can have streamUrls-arrays i.e containing just one element being wss.

Still, since you prefer the tidy approach, let's go for that :)